### ISSUE TITLE
LambdaPrinter and NumPyPrinter should have their own print methods

### ISSUE THREAD

**Issue by asmeurer** [sympy/sympy#11023]
Right now LambdaPrinter and NumPyPrinter just inherit the print method `_sympystr` from StrPrinter. They should have their own printers, so that it's easier to define lambdify behavior for custom subclasses.

**Comment by moorepants** [comment]
I'd really like to see a PythonPrinter that prints code that can be evaluated numerically in Python. We have a "PythonPrinter" but it just prints the SymPy commands that you'd type in Python, as opposed to using the math module or numpy like the LambdaPrinter does. The word "lambdify" is confusing too.

Do we actually have a NumPyPrinter?

**Comment by asmeurer** [comment]
Yeah, NumPyPrinter is a subclass of LambdaPrinter (which I think is actually the Python printer you are looking for). Really all the lambdify "translations" ought to be implemented in the NumPyPrinter, so that it's just code generation. But that's a separate issue. 

Also, I just realized that this can also be achieved by implementing `_imp_` on the custom subclass.

**Comment by moorepants** [comment]
Yeah, I can raise another issue.

**Comment by asmeurer** [comment]
Also, when implementing this, we may should implement a superclass fallback, for backwards compatibility.
